### PR TITLE
feat(skyrim-platform): support array return values in scripting functions

### LINKED ISSUE #448: Support Papyrus arrays as a return value

### PR DESCRIPTION
**Pull request by shevernitskiy**
closes #448

### CODE REVIEW

**Comment by nic11** [comment]
@shevernitskiy Hey, is this about returning arrays or passing as arguments too?

**Comment by shevernitskiy** [comment]
It's just returning arrays. In ts methods with array return were throwing exceptions before. I made it work.

File: skyrim-platform/src/platform_se/skyrim_platform/CallNative.cpp [hunk 1]
```diff
@@ -111,45 +111,53 @@ class VariableAccess : public RE::BSScript::Variable
 };
 
 namespace {
-CallNative::AnySafe VariableToAnySafe(
+
+CallNative::ObjectPtr GetOneObjectPtr(
```
Comment (nic11) [inline skyrim-platform/src/platform_se/skyrim_platform/CallNative.cpp:115]: Really minor stuff, but I feel like `GetSingleObjectPtr` would be a bit better here (one/many vs single/multiple). Not really a request to change

**Review by nic11 — CHANGES_REQUESTED** [review]
Thanks for the PR! Primarily minor styleguide changes, and we're good to go, I believe.

And please could you also create a file like this describing your changes? Example: https://github.com/skyrim-multiplayer/skymp/blob/2724bfe963a62284a72a3eef9b89ba8c57632363/docs/release/dev/sp-browser-getters.md

Ideally with a short example with a TS function and some Papyrus code that uses the return value.

**Comment by nic11** [comment]
BTW Windows build failing on upload-artifact step isn't your fault, just to be clear, so I guess we can merge that when other issues are fixed. @Pospelove, are you okay with this?

**Comment by mrowrpurr** [comment]
Related Issue: https://github.com/skyrim-multiplayer/skymp/issues/448

File: skyrim-platform/src/platform_se/skyrim_platform/CallNative.cpp (hunk 1 above)
Comment (shevernitskiy) [reply skyrim-platform/src/platform_se/skyrim_platform/CallNative.cpp:115]: Do you mean place it after `VariableToAnySafe`?

**Comment by nic11** [comment]
Thanks @mrowrpurr, updated the description to state that this PR relates to that

File: skyrim-platform/src/platform_se/skyrim_platform/CallNative.cpp (hunk 1 above)
Comment (nic11) [reply skyrim-platform/src/platform_se/skyrim_platform/CallNative.cpp:115]: Nope, placement is totally ok here (at least from my point of view). I meant just renaming:

```suggestion
CallNative::ObjectPtr GetSingleObjectPtr(
```

IMO 'Single' sounds kinda more appropriate. But that's questionable

File: skyrim-platform/src/platform_se/skyrim_platform/CallNative.h [hunk 2]
```diff
@@ -32,7 +32,9 @@ class Object
 };
 
 using ObjectPtr = std::shared_ptr<Object>;
-using AnySafe = std::variant<ObjectPtr, double, bool, std::string>;
+using AnySafe = std::variant<ObjectPtr, double, bool, std::string, std::vector<double>,
+                std::vector<std::string>, std::vector<bool>,
+                std::vector<CallNative::ObjectPtr>>;
```
[the remark this answers is not in the document]
Comment (shevernitskiy) [reply skyrim-platform/src/platform_se/skyrim_platform/CallNative.h:37]: Unfortunatelly, i need help here. What is clang-format?:) Sorry, know nothing about all dev stuff and need your guidance.

File: skyrim-platform/src/platform_se/skyrim_platform/CallNative.cpp (hunk 1 above)
Comment (shevernitskiy) [reply skyrim-platform/src/platform_se/skyrim_platform/CallNative.cpp:115]: Oh.. okey, ofccourse...

File: skyrim-platform/src/platform_se/skyrim_platform/CallNative.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (nic11) [reply skyrim-platform/src/platform_se/skyrim_platform/CallNative.h:37]: Well... I guess you use MS Visual Studio, just use format document or whatever somewhere in menus

[the remark this answers is not in the document]
Comment (nic11) [reply skyrim-platform/src/platform_se/skyrim_platform/CallNative.h:37]: https://stackoverflow.com/questions/5755942/how-do-you-auto-format-code-in-visual-studio

**Comment by shevernitskiy** [comment]
[sp-array-returns.md](https://github.com/skyrim-multiplayer/skymp/files/7696775/sp-array-returns.md)
Here is a little doc, like you are requested.

**Comment by nic11** [comment]
Thanks! Added this to the PR

**Comment by nic11** [comment]
Also fixed some minor remaining stuff, so I won't bother you too much :)

**Review by nic11 — DISMISSED** [review]
LGTM

**Review by Pospelove — APPROVED** [review]

**Pull request merged**